plugin/src/withFlipper.ios.ts

```diff
@@ -0,0 +1,148 @@
+/*
+exports the following config file changes for iOS Apps
+
+withFlipper
+  (1) Inside use_react_native, after :fabric_enabled => flags[:fabric_enabled],
+  :flipper_configuration => FlipperConfiguration.enabled,
+  :flipper_configuration => FlipperConfiguration.disabled,
+
+  Depending on the ENV variable
+
+withProductionEnv
+  (1) BEFORE target line
+  production = ENV["PRODUCTION"] == "1"
+
+  (2) Inside use_react_native, after :path => config[:reactNativePath],
+  :production => production,
+*/
+
+import { withDangerousMod } from "@expo/config-plugins";
+import {
+  mergeContents,
+  removeContents,
+} from "@expo/config-plugins/build/utils/generateCode";
+import { ExpoConfig } from "@expo/config-types";
+import {
+  EXPO_FLIPPER_TAG,
+  IOS_HAS_FLIPPER_ARG,
+  IOS_HAS_PRODUCTION_ARG,
+  IOS_URN_ARG_ANCHOR,
+} from "./constants";
+import { FlipperConfig } from "./types";
+import path from "path";
+import fs from "fs";
+
+/** Create a namepaced tag */
+const tag = (s: string) => `${EXPO_FLIPPER_TAG}-${s}`;
+
+/** Return the flipper enabling line for use inside of use_react_native */
+const createFlipperArgument = (version?: string) => {
+  const active = version
+    ? `FlipperConfiguration.enabled(["Debug"], { 'Flipper' => '${version}' }),`
+    : `FlipperConfiguration.enabled,`;
+  const inactive = `FlipperConfiguration.disabled`;
+  return `:flipper_configuration => ENV['FLIPPER_DISABLE'] ? ${inactive} : ${active}`;
```

Comment:
Does this mean by default it should be installing flipper cocoapods? Or do we have to specify FLIPPER_DISABLE=0 in our environment variable?

Reply:
It appears from the react-native code that as of 0.69.2+ Flipper's always included by default, but is included conditionally inside of `use_react_native` instead of as a separate `use_flipper` directive

Reply:
One thing I'll probably do is change this to a hard-check for == "1" similar to the production check that's also needed as part of the 0.69.2 changes.

Reply:
Cool. So 

```
```

Reply:
An important note: If you're doing a production build of react-native, flipper is automatically disabled by react-native and should be skipped. This is the other reason we needed to ensure the production flag was passed in to use_react_native

Reply:
Yea. I'd assume we want to be able have FLIPPER_DISABLE=1 only in staging and production expo pipeline as a secret. And locally we woudldn't have to assign this environment variable anything to get flipper to build with flipper

Reply:
> Cool. So this part of the code should technically be automatically uncommented from expo-community-flipper plugin

Weirdly, no. When we expo prebuild, the expo template has the old flipper code from pre 0.69.2, which was the source of a lot of the SDK 46 pain. https://github.com/react-native-community/react-native-template-typescript/blob/main/template/ios/Podfile is the Podfile we want to reference for enabling Flipper, since its based on the latest RN code.

Reply:
So in addition to FLIPPER_DISABLE=1 I'd have to ensure that in our expo build pipeline that PRODUCTION=1 is also set as an environment variable?